### PR TITLE
Add notification message if job failed for ondemand pipelines.

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Add notification message if job failed for ondemand pipelines.

## How can this be tested?

Check slack channel.

### CODE REVIEW

File: .github/workflows/e2e-tests-ondemand.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
       max-parallel: 4
       matrix:
         include:
-          - version: 1-24
+          - version: 1-242
```
Comment (andriisoldatenko) [inline .github/workflows/e2e-tests-ondemand.yaml:29]:
```suggestion
          - version: 1-24
```

Comment (albertogdd) [reply .github/workflows/e2e-tests-ondemand.yaml:29]: Yes pls

File: .github/workflows/e2e-tests-ondemand.yaml [hunk 2]
```diff
@@ -67,3 +67,21 @@ jobs:
...
+    if: ${{ failure() || cancelled() }}
+    runs-on:
+      - self-hosted
+      - operator-e2e
+    steps:
+      - name: Notify failure in Slack
+        uses: slackapi/slack-github-action@70cd7be8e40a46e8b0eced40b0de447bdb42f68e # v1.26.0
+        with:
+          channel-id: 'int-cp-operator'
+          slack-message: ":x: E2E ondemand tests failed `${{ matrix.platform }}-${{ matrix.platform }}` on ${{ github.event.inputs.target }} branch (${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
```
Comment (andriisoldatenko) [inline .github/workflows/e2e-tests-ondemand.yaml:85]:
```suggestion
          slack-message: ":x: E2E ondemand tests failed `${{ matrix.platform }}-${{ matrix.version }}` on ${{ github.event.inputs.target }} branch (${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})"
```

File: .github/workflows/e2e-tests-ondemand.yaml [hunk 3]
```diff
@@ -67,3 +67,21 @@ jobs:
...
+
+
+  notify-failure:
+    name: Notify failure in Slack
+    environment: E2E ondemand
+    needs: [ "run-matrix" ]
+    if: ${{ failure() || cancelled() }}
+    runs-on:
+      - self-hosted
+      - operator-e2e
```
Comment (chrismuellner) [inline .github/workflows/e2e-tests-ondemand.yaml:79]: does this need to run on our github runners?

**Comment by chrismuellner** [comment]
The PR title is `Add notification message`. What is the notification for?

File: .github/workflows/e2e-tests-ondemand.yaml (hunk 3 above)
Comment (andriisoldatenko) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: @chrismuellner good catch, I've copy/pasted it from https://github.com/dynatrace/dynatrace-operator/blob/bf7b156866c93042a5bf32838136f10dec143162/.github/workflows/e2e-tests.yaml#L70-L72

cc @albertogdd

**Comment by andriisoldatenko** [comment]
> The PR title is `Add notification message`. What is the notification for?

@chrismuellner done.

File: .github/workflows/e2e-tests-ondemand.yaml (hunk 3 above)
Comment (albertogdd) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: Good point, it shouldn't be needed

Comment (andriisoldatenko) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: but which runner is correct ?
```
runs-on: ubuntu-latest
```
?

Comment (albertogdd) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: I think you don't even need to specify it if you don't need anything specific

Comment (chrismuellner) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: let's use the same runner as other places, which should be `ubuntu-latest`

Comment (albertogdd) [reply .github/workflows/e2e-tests-ondemand.yaml:79]: Linux always

Comment (andriisoldatenko) [reply .github/workflows/e2e-tests-ondemand.yaml]: fixed in 4204ad8c

**Review by albertogdd — APPROVED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**